aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -687,6 +692,250 @@ public void onError(Exception error) {
         );
     }
 
+    @Override
+    public void fetchUserAttributes(
+            @NonNull Consumer<List<AuthUserAttribute>> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+        awsMobileClient.getUserAttributes(new Callback<Map<String, String>>() {
+            @Override
+            public void onResult(Map<String, String> result) {
+                List<AuthUserAttribute> userAttributes = new ArrayList<>();
+                for (Map.Entry<String, String> entry : result.entrySet()) {
+                    userAttributes.add(new AuthUserAttribute(
+                            new AuthUserAttributeKey(entry.getKey()), entry.getValue()));
+                }
+                onSuccess.accept(userAttributes);
+            }
+
+            @Override
+            public void onError(Exception error) {
+                onError.accept(new AuthException(
+                        "Failed to fetch user attributes since user is not authenticated",
```

Comment:
Hmm - this message is not always accurate. I would remove the "since user is not authenticated" piece since there could be other reasons it fails (e.g. no internet) and then per Jameson's suggestion set the suggestion parameter below to "Ensure that you are logged in and online." instead of "See attached exception..."

Reply:
Make sense!